### PR TITLE
Refactor spatial join tests

### LINKED ISSUE #1006: Port point_quadtree tests to header-only API

### LINKED ISSUE #1007: Port join_quadtree_and_bounding_boxes tests to header-only API

### PR DESCRIPTION
**Pull request by harrism**
## Description
Closes #1006
Closes #1007

I realized that the `join_quadtree_and_bounding_boxes()` tests are redundant to the `quadtree_point_in_polygon` tests and `quadtree_point_to_nearest_linestring` tests since the "small" test is duplicated in those. So I eliminated that test and added expectations for `join_quadtree_and_bounding_boxes` in those tests.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/indexing/point_quadtree_test.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
```
Comment (trxcllnt) [inline cpp/tests/indexing/point_quadtree_test.cpp:2]:
```suggestion
 * Copyright (c) 2023, NVIDIA CORPORATION.
```

**Comment by trxcllnt** [comment]
I think the reason these were split out was so `join_quadtree_and_bounding_boxes()` tests could be compiled independently of the other spatial join routines, since the compile times for all of them are rather large. It also means `join_quadtree_and_bounding_boxes` can't be tested if there's a compile error in quadtree PIP/P2NS.

**Review by robertmaynard — APPROVED** [review]
Approving CMake changes

**Review by isVoid — COMMENTED** [review]
I see that tests are incremental checkpoints for developers to validate the correctness of the function. It would be nice to leave some comments in the test block for `bounding_boxes` API and mention it's a checkpoint for this API.

**Review by trxcllnt — APPROVED** [review]

File: cpp/tests/experimental/join/join_quadtree_and_bounding_boxes_test.cu [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+template <typename T>
+struct JoinQuadtreeAndBoundingBoxesErrorTest : public cuspatial::test::BaseFixture {
+};
+
+using TestTypes = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(JoinQuadtreeAndBoundingBoxesErrorTest, TestTypes);
+
+TYPED_TEST(JoinQuadtreeAndBoundingBoxesErrorTest, test_empty)
```
[the remark this answers is not in the document]
Comment (harrism) [reply cpp/tests/experimental/join/join_quadtree_and_bounding_boxes_test.cu:36]: Most of the error tests also test that there is no error for empty inputs.

[the remark this answers is not in the document]
Comment (harrism) [reply cpp/tests/experimental/join/join_quadtree_and_bounding_boxes_test.cu:36]: Comments added.
